src/main/java/seedu/address/logic/parser/ImportCommandPreparer.java

```diff
@@ -0,0 +1,124 @@
+        }
+
+        BufferedReader br = new BufferedReader(fr);
+
+        try {
+            String line = br.readLine();
+            while (line != null) {
+                String[] attributes = line.split(",", -1);
+                
+                if(attributes[0].equalsIgnoreCase("Name") ||
+                attributes[0].equalsIgnoreCase("Name:")){ // ignore headers
+                    line = br.readLine();
+                    continue;
+                }
+
+                int numAttributes = attributes.length;
+                
+                if(hasInvalidContact(attributes, numAttributes)) {
+                    hasContactWithoutName = true;
+                    line = br.readLine();
+                    continue;
+                }
+                
+                Name name = null;
+                Optional<Phone> phone = Optional.empty();
+                Optional<Email> email = Optional.empty();
+                Optional<Address> address = Optional.empty();
+                
+                try {
+                    name = ParserUtil.parseName(attributes[0]);
+                    if (!attributes[1].matches("")) {
+                        phone = Optional.of(ParserUtil.parsePhone(attributes[1]));
+                    }
+                    if (!attributes[2].matches("")) {
+                        email = Optional.of(ParserUtil.parseEmail(attributes[2]));
+                    }
+                    if (!attributes[3].matches("")) {
+                        address = Optional.of(ParserUtil.parseAddress(attributes[3]));
+                    }
+                } catch (ParseException pe) {
+                    hasContactWithInvalidField = true;
+                    line = br.readLine();
+                    continue;
+                }
+
+                ArrayList<String> tags = new ArrayList<>();
+                //Check for tags 
+                if (numAttributes >= 5) {
+                    for (int i = 4; i < numAttributes; i++) {
+                        if (!attributes[i].matches("")) {
+                            tags.add(attributes[i]);
+                        }
+                    }
+                }
+                
+                Set<Tag> tagList = ParserUtil.parseTags(tags);
+                persons.add(new Person(name, phone, email, address, tagList));
+                line = br.readLine();
+            }
+            br.close();
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
+        }
+        return new ImportCommand(persons, hasContactWithInvalidField, hasContactWithoutName);
+    }
+    /**
+        this method checks if the compulsory name field is filled up
+        Note: it is okay if the fields are filled up improperly, ie an email without @,
+        The format checking routine associated with AddCommand will handle that and alert the user
+    */
+    private boolean hasInvalidContact(String[] attributes, int numAttributes) {
```

Comment:
The method name is a bit deceiving since the contact can still be invalid given this returns true; use method name "hasNoName" or something more indicative. Also, take note that your javadoc first sentence should end with a "." .